### DISCUSSION TITLE
Is it possible to generate Lab testing for every patient?

### DISCUSSION THREAD

**Discussion by mduna** [synthetichealth/synthea#1022] [Q&A] [answered]
In my module, I have a medication and a lab. It generated the medication for every patient but not lab testing for every patient. What I did wrong? Any help will be appreciated.

**Comment by jawalonoski** [comment, marked as answer]
Well, you didn't include a module, so no one can say with any certainty.

My guess is that this is related to `exporter.years_of_history`. 

See the description for that setting under https://github.com/synthetichealth/synthea/wiki/Common-Configuration#other-export-settings

Basically, _Active_ Medications (i.e., Prescriptions) and _Active_ Conditions (i.e., Diagnoses) are never filtered from patient records. However, labs (and nearly everything else) will be filtered out if they are older than `X` years, as defined by `exporter.years_of_history`.

**Reply by mduna** [reply]
Thanks. Following your suggestion, set exporter.years_of_history=0, got lab testing for every patient. Just start to learn how to build the module. See below is my module file. 

Have another question:
By using this module, it generates the lab testing with the value I wanted. Look at the FHIR bundling, it does not have the normal range of the lab testing. I would not expect to have the normal range, since I could not find a way to build. Is it possible to have both the specific lab result and its normal range at the same time? If so, could you please give me some instruction? Thanks. 

{
  "name": "nhsnlab",
  "remarks": [
    "A blank module"
  ],
  "states": {
    "Initial": {
      "type": "Initial",
      "direct_transition": "Hospital Inpatient Admission"
    },
    "Terminal": {
      "type": "Terminal"
    },
    "Hospital Inpatient Admission": {
      "type": "Encounter",
      "encounter_class": "inpatient",
      "reason": "",
      "codes": [
        {
          "system": "SNOMED-CT",
          "code": 1505002,
          "display": "Hospital admission for isolation (procedure)"
        }
      ],
      "direct_transition": "Length of Stay"
    },
    "Length of Stay": {
      "type": "SetAttribute",
      "attribute": "hospitalization_days",
      "direct_transition": "Admission Days",
      "distribution": {
        "kind": "UNIFORM",
        "round": false,
        "parameters": {
          "high": 5,
          "low": 2
        }
      }
    },
    "Admission Days": {
      "type": "SetAttribute",
      "attribute": "admission_days",
      "direct_transition": "Hospitalization Day Begin"
    },
    "Hospitalization Day Begin": {
      "type": "Counter",
      "attribute": "admission_days",
      "action": "increment",
      "direct_transition": "insullin_1"
    },
    "Hospitalization Day End": {
      "type": "Counter",
      "attribute": "hospitalization_days",
      "action": "decrement",
      "conditional_transition": [
        {
          "transition": "Hospitalization Next Day",
          "condition": {
            "condition_type": "Attribute",
            "attribute": "hospitalization_days",
            "operator": ">",
            "value": 0
          }
        },
        {
          "transition": "Determin Discharge"
        }
      ]
    },
    "Hospitalization Next Day": {
      "type": "Delay",
      "distribution": {
        "kind": "EXACT",
        "parameters": {
          "value": 1
        }
      },
      "unit": "days",
      "direct_transition": "Hospitalization Day Begin"
    },
    "Determin Discharge": {
      "type": "Simple",
      "direct_transition": "Terminal"
    },
    "delay5Min": {
      "type": "Delay",
      "distribution": {
        "kind": "EXACT",
        "parameters": {
          "value": 5
        }
      },
      "unit": "minutes",
      "direct_transition": "Glucose_2"
    },
    "Glucose": {
      "type": "Observation",
      "category": "laboratory",
      "unit": "mg/dL",
      "codes": [
        {
          "system": "LOINC",
          "code": "2339-0",
          "display": "Glucose Bld-mCnc"
        }
      ],
      "direct_transition": "delay5Min",
      "exact": {
        "quantity": 40
      }
    },
    "insullin_1": {
      "type": "MedicationOrder",
      "codes": [
        {
          "system": "RxNorm",
          "code": 259111,
          "display": "insulin lispro 25 UNT/ML / insulin lispro protamine, human 75 UNT/ML Injectable Suspension"
        }
      ],
      "direct_transition": "Glucose",
      "administration": true,
      "prescription": {
        "dosage": {
          "amount": 1,
          "frequency": 1,
          "period": 1,
          "unit": "days"
        },
        "duration": {
          "quantity": 3,
          "unit": "days"
        },
        "refills": 3
      }
    },
    "Glucose_2": {
      "type": "Observation",
      "category": "laboratory",
      "unit": "mg/dL",
      "codes": [
        {
          "system": "LOINC",
          "code": "2339-0",
          "display": "Glucose Bld-mCnc"
        }
      ],
      "direct_transition": "delay5Min_2",
      "exact": {
        "quantity": 60
      }
    },
    "delay5Min_2": {
      "type": "Delay",
      "distribution": {
        "kind": "EXACT",
        "parameters": {
          "value": 5
        }
      },
      "unit": "minutes",
      "direct_transition": "Hospitalization Day End"
    }
  },
  "gmf_version": 2
}

**Reply by jawalonoski** [reply]
> By using this module, it generates the lab testing with the value I wanted. Look at the FHIR bundling, it does not have the normal range of the lab testing. I would not expect to have the normal range, since I could not find a way to build. Is it possible to have both the specific lab result and its normal range at the same time? If so, could you please give me some instruction? Thanks.

I'm sorry, we do not currently support exporting the normal range values.

**Reply by mduna** [reply]
Thanks.
